### PR TITLE
EXSWHTEC-12 - Implement additional tests for hipIpcGetMemHandle

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Verify that unique handles are returned for consecutive calls
- Verify that unique handles are returned for separate allocations at same address
- Verify that freed memory causes an error
- Verify that out of bounds pointer will cause an error.

### CODE REVIEW

File: tests/catch/unit/device/hipIpcGetMemHandle.cc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#ifdef __linux__
```
Comment (chrispaquot) [inline tests/catch/unit/device/hipIpcGetMemHandle.cc]: Is there anything specific to Linux in this test?

Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipIpcGetMemHandle.cc]: The IPC APIs are exclusive to Linux.

**Review by chrispaquot — DISMISSED** [review]

**Comment by mangupta** [comment]
@music-dino / @mirza-halilcevic : Some tests failed on CI. Details shared separately.

**Comment by mirza-halilcevic** [comment]
> @music-dino / @mirza-halilcevic : Some tests failed on CI. Details shared separately.

PR updated, you can proceed with the merge.

**Pull request merged**